# Scenario 1: A single data scientist participating in an ML competition
MLflow setup:

- Tracking server: no
- Backend store: local filesystem
- Artifacts store: local filesystem
- The experiments can be explored locally by launching the MLflow UI.

In [1]:
import mlflow

In [3]:
print(f"tracking URI: '{mlflow.get_tracking_uri()}'")b

tracking URI: 'file:///home/kylepaul/notebooks/mlops-zoom-camp-2022/session_2/running-mlflow-exmaples/mlruns'


In [5]:
mlflow.search_experiments()

[<Experiment: artifact_location='file:///home/kylepaul/notebooks/mlops-zoom-camp-2022/session_2/running-mlflow-exmaples/mlruns/0', creation_time=1685480696852, experiment_id='0', last_update_time=1685480696852, lifecycle_stage='active', name='Default', tags={}>]

## Creating an experiment and logging a new run

In [8]:
from sklearn.linear_model import LogisticRegression
from sklearn.datasets import load_iris
from sklearn.metrics import accuracy_score

mlflow.set_experiment("my-experiment-1")

with mlflow.start_run():
    
    X, y = load_iris(return_X_y=True)
    
    params = {"C": 0.1, "random_state": 42}
    mlflow.log_params(params)
    
    model_lr = LogisticRegression(**params).fit(X,y)
    y_pred = model_lr.predict(X)
    mlflow.log_metric("accuracy", accuracy_score(y, y_pred))
    
    mlflow.sklearn.log_model(model_lr, artifact_path="models")
    print(f"default artifacts URI: '{mlflow.get_artifact_uri()}'")

default artifacts URI: 'file:///home/kylepaul/notebooks/mlops-zoom-camp-2022/session_2/running-mlflow-exmaples/mlruns/322087927344521053/3ee44b85720242238f1b9033c6b0d3e2/artifacts'


/home/kylepaul/anaconda3/lib/python3.10/site-packages/_distutils_hack/__init__.py:33: UserWarning: Setuptools is replacing distutils.
  warnings.warn("Setuptools is replacing distutils.")


In [9]:
mlflow.search_experiments()

[<Experiment: artifact_location='file:///home/kylepaul/notebooks/mlops-zoom-camp-2022/session_2/running-mlflow-exmaples/mlruns/322087927344521053', creation_time=1685481097124, experiment_id='322087927344521053', last_update_time=1685481097124, lifecycle_stage='active', name='my-experiment-1', tags={}>,
 <Experiment: artifact_location='file:///home/kylepaul/notebooks/mlops-zoom-camp-2022/session_2/running-mlflow-exmaples/mlruns/0', creation_time=1685480696852, experiment_id='0', last_update_time=1685480696852, lifecycle_stage='active', name='Default', tags={}>]

xml file created in folder 0

artifact_location: file:///home/kylepaul/notebooks/mlops-zoom-camp-2022/session_2/running-mlflow-exmaples/mlruns/0
creation_time: 1685480696852
experiment_id: '0'
last_update_time: 1685480696852
lifecycle_stage: active
name: Default

In [ ]:
xml file created in folder 1

artifact_location: file:///home/kylepaul/notebooks/mlops-zoom-camp-2022/session_2/running-mlflow-exmaples/mlruns/322087927344521053
creation_time: 1685481097124
experiment_id: '322087927344521053'
last_update_time: 1685481097124
lifecycle_stage: active
name: my-experiment-1

## interacting with the model registry

In [10]:
from mlflow.tracking import MlflowClient

client = MlflowClient()

In [13]:
from mlflow.exceptions import MlflowException

try:
    client.search_registered_models()
except MlflowException:
    print("It's not possible to access the model registry :(")